src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { Fragment, useEffect, useState } from 'react';
+import classNames from 'classnames';
+
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo;
+  isShowLoader: boolean;
+  onDelete?: (todoId: number) => void;
+  onUpdateStatus?: (todo: Todo) => void;
+  onUpdateTitle?: (todo: Todo) => Promise<void>;
+  inputRef?: React.RefObject<HTMLInputElement>;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  isShowLoader,
+  onDelete = () => {},
+  onUpdateStatus = () => {},
+  onUpdateTitle = () => Promise.resolve(),
+  inputRef,
+}) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [newTitle, setNewTitle] = useState(todo.title);
+
+  const handleSubmit = async (event: React.FormEvent) => {
```

Comment:
Let's have some consistency, if you decided to use `async/await` use it everywhere